src/providers/ldap/sdap.c

```diff
@@ -1276,6 +1277,15 @@ errno_t sdap_set_config_options_with_rootdse(struct sysdb_attrs *rootdse,
         }
     }
 
+    if (!sdom->naming_context) {
+        sdom->naming_context = talloc_strdup(sdom, naming_context);
+        if (sdom->naming_context == NULL) {
+            DEBUG(SSSDBG_IMPORTANT_INFO,
+                  "Failed to copy naming context (base DN) [%s], "
+                  "will continue with unset value.\n", naming_context);
+        }
+    }
+
```

Comment:
What if all `sdom->*_base` and `sdom->naming_context` are all set (i. e. not NULL)?

In such case local variable `naming_context` stays NULL but it is used later in the function. I just wonder if that is OK or should we rather use  `sdom->naming_context` in `sdap_set_search_base`?

Reply:
I guess it is OK, just I'm not sure...

Reply:
If they are all set, the local variable `naming_context` will not be used because every use is surrounded by a check:
```
if (!sdom->*_base) {
   (naming_context used here)
}
```

The only use outside these checks is the final `talloc_free()`, but that's OK as it is initialized to `NULL`.

Reply:
no, look further down in the function (not changed by this PR and therefore not shown in github UI. I mean this:

    if (!sdom->search_bases) {
        ret = sdap_set_search_base(opts, sdom,
                                   SDAP_SEARCH_BASE,
                                   naming_context);
        if (ret != EOK) goto done;
    }

Reply:
Ahh OK, got it. You are right.